account_invoice_supplier_self_invoice/models/account_move.py

```diff
@@ -0,0 +1,87 @@
+# © 2017 Creu Blanca
+# Copyright 2022 - Moduon
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    self_invoice_number = fields.Char(readonly=True, copy=False)
+    is_self_invoice_number_different = fields.Boolean(
+        string="Different invoice reference and self invoice number",
+        compute="_compute_is_self_invoice_number_different",
+        help="Self invoice number is different than invoice reference",
+    )
+    set_self_invoice = fields.Boolean(
+        help="Set self invoice for this invoice.",
+    )
+    can_self_invoice = fields.Boolean(related="partner_id.self_invoice")
+
+    def _compute_is_self_invoice_number_different(self):
+        self.update({"is_self_invoice_number_different": False})
+        for move in self:
+            if move.move_type not in ("in_invoice", "in_refund") or not (
```

Comment:
Remember to always use sets for `in` or `not in` operators, if possible. This is a 2-element tuple, so the performance benefit if you accept the suggestion will be ridiculous. But my inner nerd forces me to add the suggestion, sorry 😆

```suggestion
            if move.move_type not in {"in_invoice", "in_refund"} or not (
```